common/client/multi_node.go

```diff
@@ -26,159 +22,139 @@ var (
 		Name: "multi_node_states",
 		Help: "The number of RPC nodes currently in the given state for the given chain",
 	}, []string{"network", "chainId", "state"})
-	// PromMultiNodeInvariantViolations reports violation of our assumptions
-	PromMultiNodeInvariantViolations = promauto.NewCounterVec(prometheus.CounterOpts{
-		Name: "multi_node_invariant_violations",
-		Help: "The number of invariant violations",
-	}, []string{"network", "chainId", "invariant"})
 	ErroringNodeError = fmt.Errorf("no live nodes available")
 )
 
 // MultiNode is a generalized multi node client interface that includes methods to interact with different chains.
 // It also handles multiple node RPC connections simultaneously.
 type MultiNode[
 	CHAIN_ID types.ID,
-	SEQ types.Sequence,
-	ADDR types.Hashable,
-	BLOCK_HASH types.Hashable,
-	TX any,
-	TX_HASH types.Hashable,
-	EVENT any,
-	EVENT_OPS any,
-	TX_RECEIPT types.Receipt[TX_HASH, BLOCK_HASH],
-	FEE feetypes.Fee,
-	HEAD types.Head[BLOCK_HASH],
-	RPC_CLIENT RPC[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, BATCH_ELEM],
-	BATCH_ELEM any,
-] interface {
-	clientAPI[
-		CHAIN_ID,
-		SEQ,
-		ADDR,
-		BLOCK_HASH,
-		TX,
-		TX_HASH,
-		EVENT,
-		EVENT_OPS,
-		TX_RECEIPT,
-		FEE,
-		HEAD,
-		BATCH_ELEM,
-	]
-	Close() error
-	NodeStates() map[string]string
-	SelectNodeRPC() (RPC_CLIENT, error)
-
-	BatchCallContextAll(ctx context.Context, b []BATCH_ELEM) error
-	ConfiguredChainID() CHAIN_ID
-}
-
-type multiNode[
-	CHAIN_ID types.ID,
-	SEQ types.Sequence,
-	ADDR types.Hashable,
-	BLOCK_HASH types.Hashable,
-	TX any,
-	TX_HASH types.Hashable,
-	EVENT any,
-	EVENT_OPS any,
-	TX_RECEIPT types.Receipt[TX_HASH, BLOCK_HASH],
-	FEE feetypes.Fee,
-	HEAD types.Head[BLOCK_HASH],
-	RPC_CLIENT RPC[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, BATCH_ELEM],
-	BATCH_ELEM any,
+	RPC any,
```

Comment:
Same comment as above. I believe we should be explicit here and use the NodeClient instead.

Reply:
Marking this comment resolved as I replied to the other comment :)